### PR TITLE
[KernelGen][Nvidia] Add If operator with Triton kernel

### PR DESCRIPTION
**Pull request by Dingxingdi**
## Summary
Adds a Triton kernel for `If`, implementing conditional selection with `tl.where`.

## Testing
- Validated against PyTorch `torch.where` reference on device side
- Accuracy test: `CUDA_VISIBLE_DEVICES=0 python -m pytest tests/test_If.py -x -v --timeout=60`
- Benchmark: `CUDA_VISIBLE_DEVICES=0 python -m pytest benchmark/test_If.py --level core -s`
- Tested on: NVIDIA H20, CUDA 13.2, Docker container `dxd`

## Performance
Test command: `pytest benchmark/test_If.py --level core -s` (NVIDIA H20)

| Dtype | Cases | Mean Speedup |
|---|---:|---:|
| float16 | 5 | 1.475x |
| float32 | 5 | 1.442x |
| bfloat16 | 5 | 1.378x |
| **Arithmetic Mean** | **15** | **1.432x** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 1.432x | Primary |
| Tianshu | PASS | PASS | 1.004x | — |
| Muxi | FAIL | N/A | — | Existing backend result reports failure |
| Ascend | PASS | N/A | — | — |
| Hygon | PASS | PASS | 0.943x | — |

## Files Changed
- `src/flag_gems/ops/If.py`: Triton kernel implementation
- `tests/test_If.py`: Accuracy test
- `benchmark/test_If.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry

### CODE REVIEW

**Comment by w1120029931-bit** [comment]
please resolve conflicts

File: conf/operators.yaml [hunk 1]
```diff
@@ -1,4 +1,15 @@
 ops:
+  - id: If
```
Comment (w1120029931-bit) [inline conf/operators.yaml]: please follow the order

File: benchmark/test_If.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    else_val = utils.generate_tensor_input(shape, cur_dtype, device)
+    yield condition, then_val, else_val
+
+
+@pytest.mark.If
+def test_If():
+    bench = base.GenericBenchmark(
+        op_name="If",
+        input_fn=if_input_fn,
+        torch_op=flag_gems.If,
```
Comment (tengqm) [inline benchmark/test_If.py]: ?

File: conf/operators.yaml [hunk 3]
```diff
@@ -3693,6 +3693,17 @@ ops:
     stages:
       - beta: '5.0'
       - stable: '5.3'
+  - id: If
```
Comment (tengqm) [inline conf/operators.yaml]: This `If` is equivalent to `where`, right?

File: benchmark/test_If.py (hunk 2 above)
Comment (Dingxingdi) [reply benchmark/test_If.py]: Fixed. I use `torch.where` as the baseline.

File: conf/operators.yaml (hunk 3 above)
Comment (Dingxingdi) [reply conf/operators.yaml]: Yes, I think so.
So I use `torch.where` as the baseline.

File: conf/operators.yaml [hunk 4]
```diff
@@ -1,4 +1,15 @@
...
+    description: Triton kernel implementation for If.
+    for:
+      - If
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Math
+    stages:
+      - alpha: '5.1'
```
Comment (tengqm) [inline conf/operators.yaml:12]: why '5.1'?

File: conf/operators.yaml [hunk 5]
```diff
@@ -1,4 +1,15 @@
 ops:
+  - id: If
+    description: Triton kernel implementation for If.
+    for:
+      - If
+    labels:
+      - aten
```
Comment (tengqm) [inline conf/operators.yaml:7]: Are you sure this is an Torch Aten operator?
